javalib/src/main/scala/java/util/AbstractMap.scala

```diff
@@ -0,0 +1,209 @@
+
+  def clear(): Unit =
+    entrySet.clear
+
+  def keySet(): java.util.Set[K] =
+    new java.util.Set[K] {
+      override def size = self.size
+
+      override def contains(k: Any) = self.containsKey(k)
+      def containsAll(l: java.util.Collection[_]): Boolean =
+        !(for {
+          e <- l
+        } yield { self.containsKey(e) }).exists(_ == false) //To be verified
+
+      def add(x: K): Boolean = throw new UnsupportedOperationException()
+      def addAll[K1 <: K](x: java.util.Collection[K1]): Boolean = throw new UnsupportedOperationException()
+
+      def clear(): Unit = self.clear
+
+      def isEmpty(): Boolean = self.isEmpty
+
+      def iterator(): java.util.Iterator[K] = throw new UnsupportedOperationException()
+
+      def remove(x: Any): Boolean = self.remove(x) != null
+      def removeAll(l: java.util.Collection[K]): Boolean =
+        !(for {
+          e <- l
+        } yield { self.remove(e) != null }).exists(_ == true)
+
+      def retainAll(x: java.util.Collection[K]): Boolean = throw new UnsupportedOperationException()
+
+      def toArray[T](x: Array[T]): Array[T] = throw new UnsupportedOperationException()
+      def toArray(): Array[Any] = throw new UnsupportedOperationException()
+
+    }
+
+  def values(): java.util.Collection[V] =
+    new java.util.Collection[V] {
+      override def size = self.size
+
+      override def contains(k: Any) = self.containsKey(k)
+      def containsAll(l: java.util.Collection[_]): Boolean =
+        !(for {
+          e <- l
+        } yield { self.containsValue(e) }).exists(_ == false) //To be verified
+
+      def add(x: V): Boolean = throw new UnsupportedOperationException()
+      def addAll[V1 <: V](x: java.util.Collection[V1]): Boolean = throw new UnsupportedOperationException()
+
+      def clear(): Unit = self.clear
+
+      def isEmpty(): Boolean = self.isEmpty
+
+      def iterator(): java.util.Iterator[V] = throw new UnsupportedOperationException()
+
+      def remove(x: Any): Boolean = self.remove(x) != null //Really do this in java???
+      def removeAll(l: java.util.Collection[V]): Boolean =
+        !(for {
+          e <- l
+        } yield { self.remove(e) != null }).exists(_ == true)
+
+      def retainAll(x: java.util.Collection[V]): Boolean = throw new UnsupportedOperationException()
+
+      def toArray[T](x: Array[T]): Array[T] = throw new UnsupportedOperationException()
+      def toArray(): Array[Any] = throw new UnsupportedOperationException()
+
+    }
+
+  override def equals(o: Any): Boolean =
+    if (o == this) true
+    else {
+      o match {
+        case m: Map[K, V] =>
+          if (self.size == m.size) {
+            !(for {
+              e <- entrySet
+            } yield {
+              m.get(e.getKey) == e.getValue
+            }).exists(_ == false)
+          } else false
+        case _ => false
+      }
+    }
+
+  override def hashCode(): Int =
+    (for {
+      e <- entrySet
+    } yield {
+      e.hashCode
+    }).sum
```

Comment:
You most likely want something like `scala.util.hashing.MurmurHash3.orderedHash`